src/pdc-api.ts

```diff
@@ -0,0 +1,79 @@
+import { client } from './client';
+import { logger } from './logger';
+import type { AccessTokenSet } from './oidc';
+
+const callPdcApi = async <T>(
+  baseUrl: string,
+  path: string,
+  params: Record<string, string>,
+  token: AccessTokenSet,
+  method: 'get' | 'post',
+  data?: unknown,
+): Promise<T> => {
+  const url = new URL(path, baseUrl);
+  url.search = new URLSearchParams(params).toString();
+  const response = await client.request<T>(
+    {
+      method,
+      url: url.toString(),
+      headers: {
+        authorization: `Bearer ${token.access_token}`,
+      },
+      data,
+    },
+  );
+  return response.data;
+};
+
+interface ApiBaseField {
+  id: number;
+  label: string;
+  shortCode: string;
+}
+
+const getBaseFields = (baseUrl: string, token: AccessTokenSet) => (
+  callPdcApi<ApiBaseField[]>(
+    baseUrl,
+    '/baseFields',
+    {},
+    token,
+    'get',
+  )
+);
+
+interface ApiProposal {
```

Comment:
Having re-written these interfaces in some of the other scripts, I think we might be at the point of benefiting from publishing a `@types/pdc` package from the `service` repo and importing those types here.

Reply:
We might at some point actually consider releasing a full blown SDK / client for the service API (which would include types)

Reply:
Does OpenAPI / Swagger help with this? In theory it should be able to automatically generate such SDKs for multiple languages, but I haven't actually looked in to using that in practice.

Reply:
Oooooh that'd be awesome.  Here are some resources related to that:

- https://github.com/OpenAPITools/openapi-generator
- https://github.com/swagger-api/swagger-codegen

Worth exploring, and probably good enough for our context.  More broadly though there are a few different ways to implement an SDK (e.g. functional vs object oriented) so I'm pretty curious if the quality / tree shaking potential of the resulting SDKs are comparable to a home brewed one!

Reply:
It would still be nice to have a single definition even within the `service` project where either the API spec gets generated from the original interfaces or vice versa. But I digress.

Reply:
Yeah, no matter what approach we take, I don't think it's going to get done this month or as part of this PR.

I opened https://github.com/PhilanthropyDataCommons/service/issues/408 to track this issue.